docs/cookbook/rag.md

```diff
@@ -0,0 +1,257 @@
+# Retrieval Augmented Generation (RAG)
+
+When you need to provide additional information to your model that it hasn't been pretrained on, we retrieve the relevant information from an external source and provide it as context to our model.
+
+## A Barebones Retrieval Example
+
+!!! note
+
+    Take a look at the [code](https://github.com/Mirascope/mirascope/blob/main/cookbook/basic_rag_example/local_embeddings.py) in our repo.
+
+Here are four simple statements about the 2022/3 soccer season that our chat model does not know:
+
+```python
+soccer_info = [
+    "Manchester City won the league, continuing their domestic dominance.",
+    "In La Liga, FC Barcelona emerged victorious with a sturdy defense.",
+    "Dortmund slipped up in the last matchday to hand Bayern the Bundesliga title.",
+    "For the first time since Maradona's days, Napoli lifted the Serie A trophy.",
+]
+```
+
+The end goal is to be able to ask our model: which team in {insert_country} won the league in the 2023 season? Then, we want to put this additional context in our prompt to obtain the correct answer.
+
+### Embeddings
+
+To pick the text most relevant to our question from our information pool, we first need to embed each of the statements in `soccer_info`. Through our wrapper class [`OpenAIChat`](../api/chat/models.md#mirascope.chat.models.OpenAIChat), we can access the `OpenAI` model through the `OpenAI.client` attribute, then call [OpenAI's embeddings](https://platform.openai.com/docs/guides/embeddings):
+
+```python
+from mirascope import OpenAIChat, Prompt
+
+chat = OpenAIChat(api_key="YOUR_OPENAI_API_KEY")
+embeddings_model = "text-embedding-ada-002"
+
+embeddings = [
+    chat.client.embeddings.create(model=embeddings_model, input=[text])
+    .data[0]
+    .embedding
+    for text in soccer_info
+]
+
+df = pd.DataFrame({"texts": soccer_info, "embeddings": embeddings})
```

Comment:
it would be useful for every step of the way to show the results of each block. this helps the user follow the flow

Reply:
Strong agree. While not necessary in the code example, seeing the output here as though it were run in Jupyter can be extremely helpful. Some of the concept pages have examples of this I believe.